src/UploadInterface.js

```diff
@@ -1,51 +1,161 @@
+  const [images, setImages] = useState([]);
+  const [userMessage, setUserMessage] = useState('');
+  const [errorOccurred, setErrorOccurred] = useState(false);
+  const [loading, setLoading] = useState(false);
   const imgPreviewEl = useRef(null);
 
+  async function uploadBtnClicked() {
+    setUserMessage('MyPhotoHub is validating user token...');
+    setLoading(true);
+    setErrorOccurred(false);
+
+    const octokit = new Octokit({ auth: token });
+    let username, repoCreated;
+    try {
+      username = await getUser(octokit);
+    } catch (err) {
+      console.error('Invalid PAT token');
+      setUserMessage('Sorry, could not find a user with the token');
+      setErrorOccurred(true);
+      setLoading(false);
+      return;
+    }
+
+    setUserMessage('User validated! MyPhotoHub is creating a GitHub repository...');
+    repoCreated = await createRepo(octokit, username, repository);
+
+    if (repoCreated) {
+      setUserMessage('GitHub repository has been created! Creating optimized images...');
+
+      const convertedFiles = await createOptimizedImages(images);
+      if (!convertedFiles) {
+        setUserMessage('Something went wrong while converting images...');
+        setLoading(false);
+        return;
+      }
+      const pathNames = getPathNamesFromFile(convertedFiles);
+
+      setUserMessage('Images optimized! Preparing image files for a new commit...');
+      try {
+        const listOfBlob = await createBlobForFile(octokit, username, repository, convertedFiles);
+
+        const currentCommit = await getCurrentCommit(octokit, username, repository, 'main');
+
+        const newTree = await createNewTree(
+          octokit,
+          username,
+          repository,
+          listOfBlob,
+          pathNames,
+          currentCommit.treeSha
+        );
+
+        setUserMessage('Now creating a commit to repo...');
+        const commitMessage = 'Add my photos to repo';
+        const newCommit = await createNewCommit(
+          octokit,
+          username,
+          repository,
+          commitMessage,
+          newTree.sha,
+          currentCommit.commitSha
+        );
+
+        await setBranchToCommit(octokit, username, repository, 'main', newCommit.sha);
+
+        setUserMessage(
+          `Process complete. You can check out https://github.com/${username}/${repository}`
+        );
+        setLoading(false);
+      } catch (err) {
+        setLoading(false);
+        setErrorOccurred(true);
+        setUserMessage(`Something went wrong while creating a commit`);
+        console.error('Error occurred while creating a commit');
+      }
+    } else {
+      console.error('Error occurred while creating a repo');
+      setUserMessage('Sorry, something went wrong while creating a repository');
+      setErrorOccurred(true);
+      setLoading(false);
+    }
+  }
+
   return (
     <div className="App">
       <header className="preview">
+        {!userMessage ? (
+          <></>
```

Comment:
`null` is more common when you want to show nothing.

Reply:
Will be addressed